stanford_media.post_update.php

```diff
@@ -0,0 +1,355 @@
+      $form_display->setComponent('field_media_oembed_video');
+      $form_modified = TRUE;
+    }
+
+    if ($form_modified) {
+      $form_display->save();
+    }
+  }
+
+  foreach (EntityViewDisplay::loadMultiple() as $view_display) {
+    if ($view_display->getTargetEntityTypeId() == 'media' && $view_display->getTargetBundle() == 'video') {
+      $view_display->removeComponent('field_media_video_embed_field');
+      $view_display->setComponent('field_media_oembed_video', [
+        'type' => 'oembed',
+        'label' => 'hidden',
+      ]);
+      $view_display->save();
+    }
+  }
+
+  drupal_flush_all_caches();
+
+  $source_config = [
+    'source_field' => 'field_media_oembed_video',
+    'providers' => ['YouTube', 'Vimeo'],
+    'thumbnails_directory' => 'public://oembed_thumbnails',
+  ];
+  /** @var \Drupal\media\MediaTypeInterface $media_type */
+  $media_type = MediaType::load('video');
+  $media_type->set('source_configuration', $source_config);
+  $media_type->set('source', 'oembed:video');
+  $media_type->calculateDependencies();
+  $media_type->save();
+
+}
+
+/**
+ * Update text formats by switching the media buttons.
+ */
+function stanford_media_post_update_8201() {
+
+  /** @var \Drupal\editor\EditorInterface $editor */
+  foreach (Editor::loadMultiple() as $editor) {
+
+    $settings = $editor->getSettings();
+    foreach ($settings['toolbar']['rows'] as &$row_items) {
+      foreach ($row_items as &$row_group) {
+        if ($media_browser_position = array_search('media_browser', $row_group['items'])) {
+          $row_group['items'][$media_browser_position] = 'DrupalMediaLibrary';
+          $editor->setSettings($settings);
+          $editor->save();
+        }
+      }
+    }
+  }
+
+  $display_modes = ['full'];
+  $media_settings = \Drupal::configFactory()
+    ->getEditable('stanford_media.settings');
+  foreach ($media_settings->get('embeddable_image_styles') as $image_style) {
+    $display_modes[] = _stanford_media_post_update_8202_image_display_mode($image_style);
+  }
+  $display_modes = array_unique($display_modes);
+  $media_settings->delete();
+
+  /** @var \Drupal\filter\FilterFormatInterface $filter */
+  foreach (FilterFormat::loadMultiple() as $filter) {
+    $filters = $filter->get('filters');
+    if (isset($filters['entity_embed'])) {
+      $filters['media_embed'] = [
+        'id' => 'media_embed',
+        'provider',
+        'media',
+        'status' => TRUE,
+        'weight' => $filters['entity_embed']['weight'],
+        'settings' => [
+          'default_view_mode' => 'full',
+          'allowed_view_modes' => array_combine($display_modes, $display_modes),
+        ],
+      ];
+      // Make sure if the filter strips html tags, we need to keep the
+      // drupal-media tag and its attributes.
+      if (isset($filters['filter_html'])) {
+        $filters['filter_html']['settings']['allowed_html'] .= ' <drupal-media data-entity-type data-entity-uuid data-align data-caption data-* alt>';
```

Comment:
This could be simplified to `<drupal-media data-* alt>`

Reply:
see https://github.com/SU-SWS/stanford_text_editor/pull/7#discussion_r334615775

Reply:
Gotcha, perhaps add a comment so the next dev doesn't try to simplify.

Reply:
well it the update would work. So if i were to simplify it, nothing would break. Then if a person edits the filters via the UI, they'd get a validation error that points out exactly the same thing. And this only runs one time so :man_shrugging: